packages/dev/core/src/Materials/Node/Blocks/PBR/refractionBlock.ts

```diff
@@ -311,27 +312,29 @@ export class RefractionBlock extends NodeMaterialBlock {
 
         state._emitUniformFromString(this._refractionMatrixName, NodeMaterialBlockConnectionPointTypes.Matrix);
 
-        state._emitFunction(
-            "sampleRefraction",
-            `
-            #ifdef ${this._define3DName}
-                #define sampleRefraction(s, c) textureCube(s, c)
-            #else
-                #define sampleRefraction(s, c) texture2D(s, c)
-            #endif\n`,
-            `//${this.name}`
-        );
-
-        state._emitFunction(
-            "sampleRefractionLod",
-            `
-            #ifdef ${this._define3DName}
-                #define sampleRefractionLod(s, c, l) textureCubeLodEXT(s, c, l)
-            #else
-                #define sampleRefractionLod(s, c, l) texture2DLodEXT(s, c, l)
-            #endif\n`,
-            `//${this.name}`
-        );
+        if (state.shaderLanguage !== ShaderLanguage.WGSL) {

```

Comment:
What happens when shader language is WGSL?

Reply:
The code has no need to differentiate between samplers as (and this is great design in webgpu:)), the type of the sampling is owned by the texture). So all calls are with textureSampler